### ISSUE TITLE
Microsoft.ApplicationInsights.TraceListener is only at 2.4.0-beta2 all other packages at ^2.4.0

### ISSUE THREAD

**Issue by sgwatgit** [Microsoft/ApplicationInsights-dotnet#588]
Hi,

I'm trying to figure out what combination of packages I should be installing. I'm build webforms apps that are being deployed to azure app services. I used the 'visual studio 2017 -> project -> right click -> Add -> Application Insights' process which installs all packages at the 2.2.0 (except for `Microsoft.ApplicationInsights.Agent.Intercept` which is installed at 2.0.6).

Since `Microsoft.ApplicationInsights` is at 2.4.0 and most other related packages have been rev'd to 2.4.x I wanted to ensure I was using the latest bits.

But `Microsoft.ApplicationInsights.TraceListener` latest release still has a `2.4.0-beta2` revision which was released [2 months ago](https://preview.nuget.org/packages/Microsoft.ApplicationInsights.TraceListener/2.4.0-beta2).

The [CHANGELOG](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/master/CHANGELOG.md) for the repository doesn't tell me much about what was updated since `2.4.0-beta1` and that seems to be for the namespace as a whole.

That documentation points me to a [release notes page](http://azure.microsoft.com/en-us/documentation/articles/app-insights-release-notes-dotnet/), that page redirects to https://docs.microsoft.com/en-us/azure/. After drilling down through the UI to Application Insights, I end up at this page https://docs.microsoft.com/en-us/dotnet/api/microsoft.applicationinsights?view=appinsights-2.3.0-beta3 which only has `2.3.0-beta3` as a selectable version and according to Nuget was [released 4 months ago](https://preview.nuget.org/packages/Microsoft.ApplicationInsights/2.3.0-beta3).

I realize there are lots of moving parts, things are in flux and documentation is HARD!

I love the https://docs.microsoft.com/en-us/ experience compared to msdn but in regards to Application Insights it seems things are in a bit of disarray.

### TLDR; I'm can't find documentation and here are my questions

How are different versions of the extensibility packages released when everything seems to have [one source repository](https://github.com/Microsoft/ApplicationInsights-dotnet)?

Is `Microsoft.ApplicationInsights.TraceListener` stable at `2.4.0-beta2`? Should I be using the last released version on Nuget, which is `2.2.0`?

Thanks!

**Comment by pharring** [comment]
@sgwatgit Application Insights actually has several repos. The Home repo is https://github.com/microsoft/ApplicationInsights-Home and the readme there includes links to the others.
Some of the repos produce more than one nuget package. For example, the TraceListener Nuget that you asked about is just one of the about comes from https://github.com/Microsoft/ApplicationInsights-dotnet-logging
Most of the nuget packages depend on the Base SDK (this repo, where you opened this issue): https://github.com/Microsoft/ApplicationInsights-dotnet as such, this Base SDK tends to stabilize before the others.
You've caught us at a time when most of the repos have stabilized on the 2.4.0 release. The logging repo, has not. I admit, though, that it has languished longer than I would like at "beta2".

@karolz-ms @cijothomas Can we publish stable versions of the logging nuget packages?

**Comment by karolz-ms** [comment]
Paul, AI NuGet publishing is beyond my control. @cijothomas can probably expedite it.

@AlexBulankou @Dmitry-Matveev might also be able to help

**Comment by sgwatgit** [comment]
@pharring Thanks for the details. That helps clarify things.

1. Is there documentation somewhere detailing these connections?

Living in both the c# and javascript worlds I have experience with nuget and npm. When using npm I can always click the link to their source repository (almost always github) and then click the releases to see when versions were released and what changes were included, both breaking and feature additions

2. Is it simply the nature of Nuget's current (not the new beta site) site design that doesn't expose these connections between the packages and the code that created them
3. Is it at the repository level that it isn't possible to easily connect created packages with source code changes?

Maybe I missed some posts on https://blogs.msdn.microsoft.com/dotnet/ and the package->repo, repo->package relationships are detailed there?

**Comment by Dmitry-Matveev** [comment]
2.2 of TraceListener is now available on [Nuget.org](https://www.nuget.org/packages/Microsoft.ApplicationInsights.TraceListener/2.2.0)
